src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/IdentifiersShouldNotContainUnderscoresTests.cs

```diff
@@ -61,10 +61,10 @@ public class DoesNotMatter
         [Fact]
         public async Task CA1707_ForNamespace_CSharp()
```

Comment:
Code fix doesn't work here. The problem seems to come from https://github.com/dotnet/roslyn-analyzers/pull/3706/files#diff-e68d732a091f5743e18f7949e45cdf47R42. In the case of the namespace that's not `GetDeclaredSymbol` but `GetSymbolInfo` that shall be called here.

Note sure if that's better to call one or the other based-upon the kind/type of the parent node or to always call `GetSymbolInfo` if `GetDeclaredSymbol` is null.

Reply:
@Evangelink 
The [docs](https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.modelextensions.getdeclaredsymbol?view=roslyn-dotnet) for GetDeclaredSymbol states the following about the node parameter:

> A syntax node that is a declaration. This can be any type derived from MemberDeclarationSyntax, TypeDeclarationSyntax, EnumDeclarationSyntax, **NamespaceDeclarationSyntax**, ParameterSyntax, TypeParameterSyntax, or the alias part of a UsingDirectiveSyntax

Is that a doc error? or I'm missing something?

I'll  wait for @mavasani to determine whether to call the correct method based on the parent node, or always call `GetDeclaredSymbol` first, then `GetSymbolInfo` if the result was null.

Reply:
Thanks for the doc! I haven't looked at it and you are right. The thing is that `node` is of type `IdentifierNameSyntax` and not `NamespaceDelcarationSyntax`, if you do `GeDeclaredSymbol(node.Parent)` that's ok.

Reply:
Alright so it's actually a little more complex, sometimes node is `IdentifierNameSyntax` and sometimes it is not (`PropertyDeclarationSyntax`). If it's an identifier you need to access the parent otherwise you should not. You will need to create an abstract method and override it in sub-classes to get the right node.